src/lib/src/component/tree-viewer/indicator.box.component.spec.ts

```diff
@@ -110,4 +110,24 @@ describe('IndicatorBoxComponent', () => {
     expect(indicatorBoxTag.nativeElement.attributes['title'].value).toEqual('Last run of test "test" was successful');
     expect(indicatorBoxTag.nativeElement.className).toEqual('fa fa-circle test-success');
   });
+
+  it('handles exceptions in condition expressions gracefully', () => {
+    // given
+    hostComponent.path = 'sample/path/to/test.tcl';
+    hostComponent.workspace = new Workspace();
+    hostComponent.workspace.setMarkerValue(hostComponent.path, 'testStatus', ElementState.Running);
+    hostComponent.workspace.setMarkerValue(hostComponent.path, 'name', 'test');
+    hostComponent.states = sampleMarkerStates.slice()
+    hostComponent.states.unshift({
+      condition: (marker) => marker.nonExisting.property === true,
+      cssClasses: 'fa fa-spinner fa-spin',
+      label: (marker) => { throw new Error('broken label provider'); },
+    });
+
+    // when
+    fixture.detectChanges()
+
+    // then
+    // expect no errors
```

Comment:
how about (additionally) expecting that the active state is false (that is the cssClasses are empty or something alike)

Reply:
My reasoning was that, if there is an exception while evaluating a state condition, it should be caught, and execution should proceed as if the condition evaluated to `false`. So, if there is another state whose condition evaluates to `true`, it will become active, and remain unaffected by other, potentially faulty states. Only if no state can become active are the defaults used.

I modified the test as you suggested, but also added another one, to distinguish those two cases.